tests/integration/components/users-cards-test.js

```diff
@@ -1,26 +1,198 @@
-    // Handle any actions with this.set('myAction', function(val) { ... });
+ test('it renders Filtered User by Team', async function(assert) {
+  this.store = this.owner.lookup('service:store');
+  this.model = this.store.createRecord('user', {
+    firstName: "Mohamad",
+    lastName: 'musaraf',
+    active: true,
+    team: 'Freshteam'
+  })
+  , this.store.createRecord('user',{ 
+    firstName: "Shalini",
+    lastName: 'Sivakumar',
+    active: true,
+    team: 'Freshsales'
+  });
+  set(this,'user',this.model)
+   // Template block usage: 
+  await render(hbs`
+    <UsersCards />
+  `);
+
+  await click('[data-test-id="filterby"]')
+  assert.dom('[data-test-id="all-teams"]').exists();
+  assert.dom('[data-test-id="Freshteam"]').hasText('Freshteam');
+  assert.dom('[data-test-id="Freshsales"]').hasText('Freshsales');
+  await click('[data-test-id="Freshteam"]');
+  assert.dom('[data-test-id="full-name"]').hasText('Mohamad musaraf');
+  assert.dom('[data-test-id="current-team"]').hasText('Freshteam');
+
+  await click('[data-test-id="filterby"]')
+  await click('[data-test-id="Freshsales"]');
+  assert.dom('[data-test-id="full-name"]').hasText('Shalini Sivakumar');
+  assert.dom('[data-test-id="current-team"]').hasText('Freshsales');
+
+  });
+
+
+  test('it renders sortBy Userlist', async function(assert) {
+    this.store = this.owner.lookup('service:store');
+    this.model = this.store.createRecord('user', {
+      firstName: "Mohamed",
+      lastName: 'Musaraf',
+      active: true,
+      team: 'Freshteam',
+      joiningDate: '2022-12-5'
+
+    })
+    , this.store.createRecord('user',{ 
+      firstName: "Shalini",
+      lastName: 'Sivakumar',
+      active: true,
+      team: 'Freshsales',
+      joiningDate: '2022-12-8'
+
+    }), this.store.createRecord('user',{ 
+      firstName: "Aswath",
+      lastName: 'Kaja',
+      active: true,
+      team: 'Freshchat',
+      joiningDate: '2022-12-7'
+
+    });
+    set(this,'user',this.model)
+     // Template block usage: 
+
+    await render(hbs`
+      <UsersCards />
+    `);
+  
+    await click('[data-test-id="sortby"]');
+    assert.dom('[data-test-id="firstName"]').hasText('First Name');
+    assert.dom('[data-test-id="lastName"]').hasText('Last Name');
+    assert.dom('[data-test-id="asc"]').hasText('Ascending');
+    assert.dom('[data-test-id="desc"]').hasText('Descending');
+
+    // Sortby FirstName
+    await click(('[data-test-id="firstName"]'));
+    assert.dom('[data-test-id="current-order-type"]').hasText('First Name');
+
+
+    await click('[data-test-id="sortby"]');
+    await click(('[data-test-id="desc"]'));
+    assert.dom('[data-test-id="full-name"]').hasText('Shalini Sivakumar');
 
-    await render(hbs`{{users-cards}}`);
+    await click('[data-test-id="sortby"]');
+    await click(('[data-test-id="asc"]'));
+    assert.dom('[data-test-id="full-name"]').hasText('Aswath Kaja');
 
-    assert.equal(this.element.textContent.trim(), '');
+    //Sortby LastName
```

Comment:
Use `:nth-child(1)` to check the ascending descending functionalities.
Check all the possible cases like what is the first child, last child, etc..